cfme/tests/cloud_infra_common/test_vm_instance_analysis.py

```diff
@@ -183,20 +166,18 @@ def _remove_docker_creds():
         view.save.click()
 
 
-@pytest.fixture(scope="module")
-def local_setup_provider(request, setup_provider_modscope, provider, appliance):
-
-    # TODO: allow for vddk parameterization
-    if provider.one_of(VMwareProvider):
-        appliance.install_vddk()
-        request.addfinalizer(appliance.uninstall_vddk)
-        set_hosts_credentials(appliance, request, provider)
-
-    if provider.one_of(EC2Provider):
-        set_agent_creds(appliance, request, provider)
+def set_host_creds(request, appliance, provider, ssa_single_vm):
+    view = navigate_to(ssa_single_vm, 'Details')
+    host_name = view.entities.summary("Relationships").get_text_of("Host")
+    host, = [host for host in provider.hosts.all() if host.name == host_name]
```

Comment:
Not a good pattern to follow:

```
>           host_data, = [data for data in provider.data['hosts'] if data['name'] == host.name]
E           ValueError: need more than 0 values to unpack
```

Reply:
I think better to use `VM` properties. like we have some REST support `properties` like `vm.host` or `vm.datastore`. It will return `host` or `datastore` object belongs to particular `vm`. 

In this case, we can go like
```host = ssa_single_vm.host```